frontend/packages/kubevirt-plugin/src/hooks/use-render-vnc-console.ts

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import { useEventListener } from './use-event-listener';
+
+interface RenderVNCConsoleArguments {
+  vmName: string;
+  shouldBeFullScreen: boolean;
+  initValue?: boolean;
+}
+
+export const useRenderVNCConsole = ({
```

Comment:
I think the implementation can be improved:

- We dont need `initValue` (as its the same as `shouldBeFullScreen`)
- you should reuse `useLocalStorage` here (it will need to be enhanced with `useEventListener(window, 'storage',..`)
- hook should return setter method for setting localstorage key value so you dont have to do things like

```js
useEventListener(window, 'beforeunload', () =>
    localStorage.removeItem(`isFullScreenVNC-${vmName}`),
  );
```
instead you would do
```
useEventListener(window, 'beforeunload', () =>
    setRenderVNCConsole(false);
  );
```